docs/content/querying/groupbyquery.md

```diff
@@ -214,3 +214,4 @@ When using the "v2" strategy, the following query context parameters apply:
 |`maxMergingDictionarySize`|Can be used to lower the value of `druid.query.groupBy.maxMergingDictionarySize` for this query.|
 |`maxOnDiskStorage`|Can be used to lower the value of `druid.query.groupBy.maxOnDiskStorage` for this query.|
 |`sortByDimsFirst`|Sort the results first by dimension values and then by timestamp.|
+|`pushDownLimit`|When a limit spec is present, enables an optimization that applies the limit at the merge buffer level, reducing the number of results that need to be processed at the broker level. Note that when the sorting order uses fields that are not in the grouping key, this can result in approximate results.|
```

Comment:
haven't gone through whole code yet... but, speaking as a user, it would be nice for druid to automatically push down limits where results would be exactly same.
and for other case, push limits only if user asked explicitly.